### DISCUSSION TITLE
Script exits without GUI

### DISCUSSION THREAD

**Discussion by ggiovannelli** [AveYo/MediaCreationTool.bat#124] [General]
Hi,

I have used this very usefull script in the past with success to create USB keys to install Win10 and Win11.

Today I have downloaded the new one, I saved from RAW as MediaCreationTool.bat but when I launch it, it opens and close suddendly and only create an Internet shortcut "latest_MCT" to "http://github.com/AveYo/MediaCreationTool.bat" .
I am not able to see the GUI anymore.

I have tried to run as administrator and normal user, by a command prompt or directly with a mouse click. Same story.
I have tried from win11 and win10 21h2 boxes.

Probably I am missing something but I am not able to understand what :)

Thanks for any tips

**Comment by AveYo** [comment]
Probably blocked by a zealous antivirus / misconfigured Defender (Controlled Folder Access)
Helps if you run it from outside your user folder (i.e. make a new folder in C:\)

**Comment by ggiovannelli** [comment]
Hi,

thanks for your kind reply.

I have found the problem :)

The problem is the EOL :)

This works:
```
::# uncomment to start auto upgrade setup directly (no prompts) - or rename script: "auto 11 MediaCreationTool.bat"^M
rem set /a AUTO=1^M
^M
::# uncomment to start create iso directly in current folder - or rename script:   "iso 20H2 MediaCreationTool.bat"^M
rem set /a ISO=1^M
^M
::# uncomment and change autodetected MediaEdition - or rename script:  "enterprise iso 2009 MediaCreationTool.bat"^M
rem set EDITION=Enterprise^M
^M
::# uncomment and change autodetected MediaLangCode - or rename script:   "de-DE home 11 iso MediaCreationTool.bat"^M
rem set LANGCODE=en-US^M
^M
::# uncomment and change autodetected MediaArch - or rename script:  "x64 iso 1909 Education MediaCreationTool.bat"^M
rem set ARCH=x64^M
^M
```

This not:
```
::# uncomment to start auto upgrade setup directly (no prompts) - or rename script: "auto 11 MediaCreationTool.bat"
rem set /a AUTO=1

::# uncomment to start create iso directly in current folder - or rename script:   "iso 20H2 MediaCreationTool.bat"
rem set /a ISO=1

::# uncomment and change autodetected MediaEdition - or rename script:  "enterprise iso 2009 MediaCreationTool.bat"
rem set EDITION=Enterprise

::# uncomment and change autodetected MediaLangCode - or rename script:   "de-DE home 11 iso MediaCreationTool.bat"
rem set LANGCODE=en-US

::# uncomment and change autodetected MediaArch - or rename script:  "x64 iso 1909 Education MediaCreationTool.bat"
rem set ARCH=x64
```

So the file to works has to be saved to Ms-Dos CR-LF not with ONLY Unix LF ....

I realized this when I saved a different copy with the simple Windows Notepad and see the size was different 123kb vs 122kb :)
Also diff finds the differences 1355 :)

```
gmarco@apps:/home/gmarco> diff -ruN MediaCreationTool.bat MediaCreationTool-broken.bat
--- MediaCreationTool.bat       2022-03-14 13:00:15.000000000 +0100
+++ MediaCreationTool-broken.bat        2022-03-12 17:15:18.000000000 +0100
@@ -1,1355 +1,1355 @@
-@goto latest at github.com/AveYo/MediaCreationTool.bat
-:Universal MCT wrapper script for all Windows 10/11 versions from 1507 to 21H2!
-:: Nothing but Microsoft-hosted source links and no third-party tools; script just configures an xml a                                       nd starts MCT
-:: Ingenious support for business editions (Enterprise / VL) selecting language, x86, x64 or AiO insid                                       e the MCT GUI
-:: Changelog: 2021.12.22
-:: - improved auto.cmd handling of mismatched OS and target edition, obey 'def', 'auto' upgrades 7 to                                        10, not 11
-:: - skip 11 checks only via auto.cmd - ignore server label, please; local acc on 11 Home; output to C                                       D or C:\ESD
-:: 11: 22000.318 / 21H2: 19044.1288 / 21H1: 19043.1348 / 20H2: 19042.1052 / 2004: 19041.572 / 1909: 18                                       363.1139
-
```

Thanks anyway for you reply.
Probably you can put in the note or in the readme to pay attention when someone save the file in a OS different from windows .

Thanks again.

**Reply by AveYo** [reply]
So you did it wrong. 
Saving the RAW file is meant to take advantage of the browser's Save as feature (`Ctrl` + `S`).
If you copy-paste the text in an already opened editor, then it's your responsibility to prepare it in advance for CRLF line endings (on windows this is a non-issue since notepad & co defaults to CRLF).
Probably should just top-right-corner `Download ZIP` at the root of the repository.
That guarantees line-endings preservation.
